src/entry.rs

```diff
@@ -616,6 +619,7 @@ impl<'a> EntryFields<'a> {
             )
         })?;
 
+        #[cfg(not(target_os = "wasi"))]
```

Comment:
Is this necessary? I think it's reasonable that if `preserve_mtime` is requested that it generates an error on wasi since it's not supported. If `preserve_mtime` is on-by-default then it can be off-by-default for wasi.

Reply:
I moved this logic into the function as you suggested and put some comments around why it is not supported.
also I implemented a default_preserve_mtime() function which turns it off by default